lib/posix-process/clone.c

```diff
@@ -476,10 +528,21 @@ static int _clone(struct clone_args *cl_args, size_t cl_args_len,
 	/* Assign the child to the scheduler */
 	uk_sched_thread_add(s, child);
 
+	/* CLONE_VFORK: Block the parent until the child calls execve()
```

Comment:
Maybe I don't fully understand the question, but I don't see a problem. `tid` is assigned by `libposixprocess` during `_uk_posix_clonetab_init()` and it's not related to whether the thread is added to the scheduler or not.

Reply:
Sorry I got confused by that `ret`. I thought that we are calling a function that returns a value and right afterwards we call a different function without even checking if that `ret` is valid or an error.... but in reality it's just a misnamed local variable... probably `tid` would have been more appropriate instead of `ret`. Anyway it's not realted to this PR I guess...